patches/minecraft/net/minecraft/client/gui/recipebook/RecipeBookGui.java.patch

```diff
@@ -9,7 +39,113 @@
           this.field_193018_j.add(new RecipeTabToggleWidget(recipebookcategories));
        }
  
-@@ -417,7 +417,7 @@
+@@ -189,24 +202,68 @@
+    }
+ 
+    private void func_193949_f() {
```

Comment:
Rather than making such a large/"messy" patch. It may make more sense to make the new logic in its own method at the bottom of the RecipeBookGui file, and then just patch calls to this method to instead call that new method.